### PR TITLE
Fix sonar issues, either by improving the code, or by suppressing the warning.

### PR DESCRIPTION
**Pull request by gklijs**
Fix sonar issues, either by improving the code, or by suppressing the warning.
fixes https://github.com/AxonFramework/extension-kafka/issues/254

### CODE REVIEW

File: kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/producer/DefaultProducerFactory.java [hunk 1]
```diff
@@ -114,16 +115,16 @@ protected DefaultProducerFactory(Builder<K, V> builder) {
             return createTransactionalProducer();
         }
 
-        if (this.nonTransactionalProducer == null) {
+        if (nonTransactionalProducer.get() == null) {
             synchronized (this) {
-                if (this.nonTransactionalProducer == null) {
+                if (nonTransactionalProducer.get() == null) {
```
Comment (smcvb) [inline kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/producer/DefaultProducerFactory.java:120]: Why not change this to an `AtomicReference#compareAndSet` operation?

**Review by smcvb — APPROVED** [review]
Looks good to me :+1:

File: kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/producer/DefaultProducerFactory.java (hunk 1 above)
Comment (gklijs) [reply kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/producer/DefaultProducerFactory.java]: Kept the 'old' code, now replaced with `updateAndGet` which seems even more fitting.

**Review by smcvb — COMMENTED** [review]
Still think this looks fine! :-)

**Pull request merged**